controller/api/destination/traffic_split_adaptor.go

```diff
@@ -65,14 +64,6 @@ func (tsa *trafficSplitAdaptor) publish() {
 			overrides = append(overrides, dst)
 		}
 		merged.Spec.DstOverrides = overrides
-	} else {
-		// If there is no traffic split, always return a destination override
```

Comment:
It looks like before this change, if there was no TrafficSplit, we would set the DstOverrides to be a single element list containing the dst.  After this change, if there is no TS and no DstOverrides are set by the user, it seems like we keep the DstOverrides as an empty list.  Is that change intentional?  Is it the behavior that we want?

Reply:
Yep, That seems right. That is what I was expecting to do. Skip the overriding when there is no TS. This means that if any `dstOverrides` were set at SP, those are propogated as is. And as you said an empty list when both are not set.

I went with the following approach because the docs about `dstOverrides` mention that **if the list is empty, requests are sent to the destination as normal**.

```go
 	// If this list is non-empty, requests to this destination should instead be
	// split between the destinations in this list.  Each destination should
	// receive a portion of the requests proportional to its weight.  If this
	// list is empty, requests should be sent to this destination as normal.
	DstOverrides []*WeightedDst `protobuf:"bytes,3,rep,name=dst_overrides,json=dstOverrides,proto3" json:"dst_overrides,omitempty"`

```

is this not the behavior in practice?

Reply:
That sounds right, but since this is a change from how we do it today, I just wanted to confirm that you've tested it and that having an empty list of DstOverrides works as expected.

Reply:
@Pothulapati Do we know how this interacts with the proxy performing a `Get` request? The default override was added to close #4879. With this change, `GetProfile` for a service will not return destination overrides so I don't think we'll perform `Get` on the endpoints now. This could lead to the proxy not performing its load balancing on the endpoints.

Reply:
So talking this through a little more we still do want to make sure `dst_overrides` for the destination profile is set if the service profile has none itself.

We should change the check here though to not be an `else` for `tsa.split != nil`. We instead would want a separate check for `if len(merged.Spec.DstOverrides) == 0`. If this is the case, we know that the traffic split added no overrides and the service profile has no overrides. Therefore, the response's `dst_overrides` should be set so that the proxy still performs its own load balancing.

Reply:
Makes Sense. Updated!